kyuubi-server/src/main/scala/org/apache/kyuubi/operation/ExecuteStatement.scala

```diff
@@ -159,7 +164,15 @@ class ExecuteStatement(
   override protected def runInternal(): Unit = {
     executeStatement()
     val sessionManager = session.sessionManager
-    val asyncOperation: Runnable = () => waitStatementComplete()
+    val asyncOperation: Runnable = () =>
+      try {
+        if (isTimeoutMonitorEnabled) {
+          addTimeoutMonitor(queryTimeout)
```

Comment:
how about moving it before `executeStatement()` as we did on the engine side?

Reply:
It's done on purpose for considering and tolerating the time for connecting and submitting statement to the engine.
But let's move it before `executeStatement` this time for semantic consistency.